From 09a9cecff738598907c0d65982c4d09925873f28 Mon Sep 17 00:00:00 2001
From: MrNerdHair <mrnerdhair@gmail.com>
Date: Sun, 17 Dec 2023 21:07:06 -0600
Subject: [PATCH] export: correctly handle paths containing whitespace, quotes,
 or nonprintable characters

In order to reliably handle paths with whitespace, quotes, or nonprintable characters, `find`, `xargs`, and `grep` should be configured to separate them using the NUL character rather than using newlines or ad-hoc escaping. (This is documented in the shellcheck linting rule SC2038, which documetns a real issue and should not be disabled.)
---
 distrobox-export | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/distrobox-export b/distrobox-export
index 77c928c21..762f2dd7b 100755
--- a/distrobox-export
+++ b/distrobox-export
@@ -347,11 +347,11 @@ export_application() {
 	# it is possible to export an app not only by its executable name but also
 	# by its launcher name.
 	desktop_files=$(
-		# shellcheck disable=SC2086,SC2038
-		find ${canon_dirs} -type f -printf "%p\n" -o -type l -printf '"%p"\n' |
-			xargs -I{} grep -le "Exec=.*${exported_app}.*" -le "Name=.*${exported_app}.*" "{}" |
-			xargs -I{} grep -Le "Exec=.*${DISTROBOX_ENTER_PATH:-"distrobox-enter"}.*" "{}" |
-			xargs -I{} printf "%s@" {}
+		# shellcheck disable=SC2086
+		find ${canon_dirs} -type f -print0 -o -type l -print0 |
+			xargs -0I{} grep -Zl -e "Exec=.*${exported_app}.*" -e "Name=.*${exported_app}.*" "{}" |
+			xargs -0I{} grep -ZL -e "Exec=.*${DISTROBOX_ENTER_PATH:-"distrobox-enter"}.*" "{}" |
+			xargs -0I{} printf "%s@" "{}"
 	)
 
 	# Ensure the app we're exporting is installed
